liblouis/compileTranslationTable.c

```diff
@@ -1402,6 +1402,11 @@ _lou_extParseChars(const char *inString, widechar *outString) {
 	for (k = 0; inString[k] && k < MAXSTRING - 1; k++) wideIn.chars[k] = inString[k];
 	wideIn.chars[k] = 0;
 	wideIn.length = k;
+	/* Fix issue #1535.
+	   For this function, the functional nature should be maintained, 
+	   i.e. the same input -> the same behavior. 
+	   Therefore, `errorCount` should not be externally influenced. */
+	errorCount = 0;
```

Comment:
I think that another issue is that `parseChars` currently always returns `1`. If it would return `0` when an error happens, the `errorCount` check could be replaced by a return value check. It would probably also fix other bugs because a return value `0` would cause other functions to terminate sooner, e.g. in https://github.com/liblouis/liblouis/blob/13cf39210defd031a798f62647ce713899079079/liblouis/compileTranslationTable.c#L4173

(`getCharacters` uses `parseChars`: https://github.com/liblouis/liblouis/blob/13cf39210defd031a798f62647ce713899079079/liblouis/compileTranslationTable.c#L1532-L1538)

Reply:
I changed `parseChars` so that it now returns 0 on error. So we no longer need to check the `errorCount` in `_lou_extParseChars`.

Reply:
Thanks!